src/main/java/io/smartcat/migration/Migration.java

```diff
@@ -45,18 +47,70 @@ public int getVersion() {
 
     /**
      * Returns migration description (for history purposes)
-     * 
-     * @return Migration description
      */
     public abstract String getDescription();
 
 	/**
      * Executes migration implementation
-     * 
-     * @throws MigrationException
+     *
+     * @throws MigrationException exception
      */
     public abstract void execute() throws MigrationException;
 
+    /**
+     * Execute provided statement and checks if the schema migration has been propagated
+     * to all nodes in the cluster. Use this method when executing schema migrations.
+     * @param statement Statement to be executed
+     * @throws MigrationException exception
+     */
+    protected void executeWithSchemaAgreement(Statement statement)
+            throws MigrationException
+    {
+        ResultSet result = this.session.execute(statement);
+        if (checkSchemaAgreement(result)) return;
+        if (checkClusterSchemaAgreement()) return;
+
+        throw new MigrationException(
+                "Failed to propagate schema update to all nodes (schema agreement error)");
```

Comment:
I ask myself should we create more specific exception for each case which will extend from `MigrationException` or use `MigrationException` throughout the code. 

I am leaning more to more specific exception so consumer can handle just for type, not go to error message.

What do you think?